code/go/internal/validator/semantic/version_integrity.go

```diff
@@ -127,3 +149,45 @@ func ensureManifestVersionHasChangelogEntry(manifestVersion string, versions []s
 	}
 	return errors.New("current manifest version doesn't have changelog entry")
 }
+
+func ensureLinksAreValid(links []string) ve.ValidationErrors {
+
+	type validateFn func(link *url.URL) error
+
+	var errs ve.ValidationErrors
+
+	validateLinks := []struct {
+		domain       string
+		validateLink validateFn
+	}{
+		{
+			"github.com",
+			validateGithub,
+		},
+	}
+	for _, link := range links {
+		linkURL, err := url.Parse(link)
+		if err != nil {
+			errs.Append(ve.ValidationErrors{
+				fmt.Errorf("invalid URL %v", err),
+			})
+			continue
+		}
+		for _, vl := range validateLinks {
+			if strings.Contains(linkURL.Host, vl.domain) {
+				if err = vl.validateLink(linkURL); err != nil {
+					errs.Append(ve.ValidationErrors{err})
```

Comment:
nit: if `validateFn` returns `ValidationErrors` you wouldn't need the transformation here